### DISCUSSION TITLE
CSS 变量模式会不会造成5.0发布会上所说的问题？

### DISCUSSION THREAD

**Discussion by taiyangzaitianshang** [ant-design/ant-design#46332] [General]
今天看更新日志发现支持css变量了。但是我记得当时发布会说，之所以5.0不选择css变量来作为样式方案是因为会产生大量的中间变量。

现在看起来貌似确实是多了许多变量，比如，原先通过Math.ceil((controlHeightLG - lineWidth - fontHeight) / 2)这种方式计算出的padding，现在会生成一个ant-design-header-vertical-padding。

所以，中间变量的膨胀其实并没有什么关系吗？

**Comment by zombieJ** [comment]
有的，在单个主题下。raw cssinjs 尺寸会比 css var 小，只有多主题混合时尺寸才会反过来。这个可以看看 @MadCcc 写的 RFC~

应用中混合主题用的越多，总体 css size 越小。相关讨论参考：

* #44654
* #45385
* [Blog 当 Ant Design 遇上 CSS 变量](https://ant-design.antgroup.com/docs/blog/css-var-plan-cn)

**Reply by taiyangzaitianshang** [reply]
好的，谢谢

**Comment by MadCcc** [comment]
还是有空间可以改善的，比如把 css 变量全部混淆掉，体积其实并不会变很多。
在多主题的场景下 css 变量模式是有体积优势的，就算不做混淆也很明显。

**Reply by MadCcc** [reply]
https://ant-design.github.io/cssinjs-benchmark/
可以通过这个 benchmark 来简单测试性能和产物体积

**Reply by taiyangzaitianshang** [reply]
好的，感谢解惑
